straxen/plugins/led_calibration.py

```diff
@@ -0,0 +1,114 @@
+'''
+Dear nT analyser, 
+if you want to complain please contact: chiara@physik.uzh.ch, gvolta@physik.uzh.ch
+'''
+
+import strax
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import numba
+from numba import njit
+
+#export, __all__ = strax.exporter()
+# QUESTION: what does 'export, __all__ = strax.exporter()' do?
+# ANSWER: [fill in]
+
+@strax.takes_config(
+    strax.Option('LED_window',
+                 default=(125, 250),
+                 help="Window (samples) where we expect the signal in LED calibration"),
+    strax.Option('noise_window',
+                 default=(350, 475),
+                 help="Window (samples) to analysis the noise"),
+    strax.Option('light_level',
+                 default=(0,248),
+                 help="Three different light level for XENON1T: (0,36), (37,126), (127,248). Defalt value: all the PMTs")
+)
+
+
+class LEDCalibration(strax.Plugin):
+    
+    '''
+    Preliminary version, several parameters to set during commisioning.
+    LEDCalibration returns: channel, time, dt, lenght, Area, amplitudeLED and amplitudeNOISE.
+    The new variables are:
+    - Area: Area computed in the given window, averaged over 6 windows that have the same starting sample and different end samples.
+    - amplitudeLED: peak amplitude of the LED on run in the given window.
+    - amplitudeNOISE: amplitude of the LED on run in a window far from the signal one.
+    '''
+    
+    __version__ = '0.0.8'
+    print(__version__)
+    depends_on = ('raw_records',)
+    # Options below copied from other plugins, need to be reviewed by an expert
+    data_kind = 'led_cal_0' 
+    compressor = 'zstd'
+    parallel = 'process'
+    rechunk_on_save = False
+    
+    dtype = [('area', np.int32, 'Area averaged in integration windows'),
+             ('amplitudeLED', np.int32, 'Amplitude in LED window'),
```

Comment:
I know I used this notation for the 1T code, but maybe we should remove the all-caps? Maybe Jelle has a stylistic preference.

Reply:
For the name of the variable you mean? I can add an underscore and avoid the capitol letters